### DISCUSSION TITLE
Tools adding Carbon Print Tests!!!

### DISCUSSION THREAD

**Discussion by MrsDaehin** [m3tomlins/performacology#6] [General]
Hey Amigos!

Last week i have seen the Carbon Print Impact showed in tools like dynatrace, and there is a experimental feature in WebPageTest to test it too!!! 🗡️ 

https://www.webpagetest.org/carbon-control/

I have planned already a WebPerf Barcelona a meetup about this 2 tools. Time to spread the word!

Can we like make a list of tools, observability, performance whatever tools we can use to measure this? IT would be great to have a list available! 

Gracias!!!

**Comment by andreig09** [comment]
I tried WebPageTest and it gives very interesting stats to have a quick status of your website. I liked the comparison with the vehicles carbon footprint.

I agree with the idea of the list of tools!

**Comment by andreig09** [comment]
Hi!

A couple of weeks ago I started a state-of-the-art review of tools that can measure carbon footprints.

Here are some conclusions and notes I've gathered so far:

- I was glad to see that now all the major cloud providers have tools to measure carbon footprints. This wasn't the case until last year.
- On the other hand, most of them are not multiplatform and don't support on-premise infrastructure.
- Also most of them are just historical reports and don't offer real-time analysis.

Tools that I found:

- [Dynatrace - Cost & Carbon Optimization](https://www.dynatrace.com/hub/detail/carbon-impact/)
- [Cloud Carbon Footprint](https://www.cloudcarbonfootprint.org/). Open-source project. Multiplatform.
- [Datadog - Hardware Sentry](https://docs.datadoghq.com/es/integrations/sentry_software_hardware_sentry/). Just for on-premise servers.
- [Azure - Emissions Impact](https://appsource.microsoft.com/en-us/product/power-bi/coi-sustainability.emissions_impact_dashboard)
- [AWS - Customer Carbon Footprint](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/what-is-ccft.html#ccft-gettingstarted)
- [Google Cloud - Carbon Footprint](https://cloud.google.com/carbon-footprint)

Cloud Carbon Footprint looks very interesting to me, I'll probably be testing the platform in the next few days.

**Reply by MrsDaehin** [reply]
hey! 

You have an option in WebPageTest to run carbon control test. It is really interesting to measure the print from your websites and services. 

As well there is a js library from which you can add a script to your online services to check the carbon print too. 

https://www.thegreenwebfoundation.org/co2-js/


i think it will be really interesting to see where these tools end up.
